src/parlant/bin/server.py

```diff
@@ -182,6 +186,49 @@ async def create_agent_if_absent(agent_store: AgentStore) -> None:
         await agent_store.create_agent(name=DEFAULT_AGENT_NAME)
 
 
+@asynccontextmanager
+async def load_modules(
+    container: Container,
+    modules: list[str],
+) -> AsyncIterator[None]:
+    if not modules:
```

Comment:
It's bad design to specifically provide a list of modules to this function, then go and do something else inside like read a file as a fallback.

This function should just get the exact modules that need to be run. And this function should **only** be run if the list of modules is non-empty.

Understanding which modules should be run based on config or params should be done in a separate function.